infra/discovery/src/listener/handler_marketplace.js

```diff
@@ -272,6 +275,56 @@ class MarketplaceEventHandler {
     }
   }
 
+  /**
+   * Store Listing and Offer events for messaging
+   * @param {Object} block
+   * @param {Object} event
+   * @param {Object} details
+   * @returns {Promise<void>}
+   * @private
+   */
+  async _storeEventForMessaging(block, event, details) {
+    const { blockNumber, logIndex } = event
+    const blockDate = new Date(block.timestamp * 1000)
+
+    const buyer = details.offer.buyer.id
```

Comment:
Just checking: you don't need to deal with the possibility that buyer/seller/party may be a proxy address, do you ?

Reply:
Yeah, totally missed that. It should always be the proxy owner. Will update it